root/static/scripts/edit/URLCleanup.js

```diff
@@ -4083,7 +4083,9 @@ const CLEANUPS: CleanupEntries = {
       },
     ],
     clean: function (url) {
-      return url.replace(/^(https?:\/\/)?((www|m)\.)?soundcloud\.com(\/#!)?/, 'https://soundcloud.com');
+      url = url.replace(/^(https?:\/\/)?((www|m)\.)?soundcloud\.com(\/#!)?/, 'https://soundcloud.com');
+      url = url.replace(/^(https:\/\/soundcloud\.com\/)(?!(?:search|tags)[\/?#])([^?]+)(?:\?.*)?$/, '$1$2');
```

Comment:
Correct but can be simplified as follows since the [quantifier](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Quantifiers) `+` is greedy (matching the maximum number of times) by default:

```suggestion
      url = url.replace(/^(https:\/\/soundcloud\.com\/)(?!(?:search|tags)[\/?#])([^?]+).*$/, '$1$2');
```